apps/sms/services/test/unit/messaging/messaging_service_test.js

```diff
@@ -0,0 +1,628 @@
+        );
+        assert.deepEqual(
+          MessagingService.mozMobileMessageClient.method.firstCall.args,
+          ['send', smsOpts.recipients[0], smsOpts.content, undefined]
+        );
+        assert.deepEqual(
+          MessagingService.mozMobileMessageClient.method.secondCall.args,
+          ['send', smsOpts.recipients[1], smsOpts.content, undefined]
+        );
+
+        results.forEach((result, idx) => {
+          assert.deepEqual(result, {
+            success: true,
+            result: successResult,
+            code: undefined,
+            recipient: smsOpts.recipients[idx]
+          });          
+        });
+      }).then(done, done);
+
+    });
+
+    test('send to one recipient unsuccessfully', function(done) {
+      smsOpts = {
+        recipients: '123',
+        content: 'hola'
+      };
+
+      MessagingService.mozMobileMessageClient.method
+        .withArgs('send').returns(Promise.reject(errorResult));
+      MessagingService.sendSMS(smsOpts).then((results) => {
+        sinon.assert.calledWithExactly(
+          MessagingService.mozMobileMessageClient.method,
+          'send', smsOpts.recipients, smsOpts.content, undefined
+        );
+
+        assert.deepEqual(results, [{
+          success: false,
+          result: undefined,
+          code: errorResult,
+          recipient: '123'
+        }]);
+      }).then(done, done);
+    });
+
+    test('send to two recipients unsuccessfully', function(done) {
+      smsOpts = {
+        recipients: ['123', '456'],
+        content: 'hola'
+      };
+
+      MessagingService.mozMobileMessageClient.method
+        .withArgs('send').returns(Promise.reject(errorResult));
+      MessagingService.sendSMS(smsOpts).then((results) => {
+        sinon.assert.calledTwice(
+          MessagingService.mozMobileMessageClient.method
+        );
+        assert.deepEqual(
+          MessagingService.mozMobileMessageClient.method.firstCall.args,
+          ['send', smsOpts.recipients[0], smsOpts.content, undefined]
+        );
+        assert.deepEqual(
+          MessagingService.mozMobileMessageClient.method.secondCall.args,
+          ['send', smsOpts.recipients[1], smsOpts.content, undefined]
+        );
+
+        results.forEach((result, idx) => {
+          assert.deepEqual(result, {
+            success: false,
+            result: undefined,
+            code: errorResult,
+            recipient: smsOpts.recipients[idx]
+          });
+        });
+      }).then(done, done);
+    });
+
+    test('send with a serviceId in dual SIM setup', function(done) {
+      this.sinon.stub(Settings, 'hasSeveralSim').returns(true);
+      smsOpts = {
+        recipients: '123',
+        content: 'hola',
+        serviceId: 0 // we use 0 because it's falsy, to test it still works
+      };
+
+      MessagingService.mozMobileMessageClient.method
+        .withArgs('send').returns(Promise.resolve(successResult));
+      MessagingService.sendSMS(smsOpts).then((results) => {
```

Comment:
Will remove the unused parameter because we already verified the results (if service method called correctly) in previous test.